### PR TITLE
MueLu, Panzer, TrilinosCouplings, Pamgen: Fix warnings

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu @trilinos/panzer @trilinos/trilinoscouplings @trilinos/pamgen 

See #13308.

### CODE REVIEW

File: packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_ReitzingerPFactory_def.hpp [hunk 1]
```diff
@@ -177,7 +177,7 @@ void ReitzingerPFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::BuildP(Level
   ArrayView<const SC> values_E, values_N;
 
   size_t Ne = EdgeMatrix->getLocalNumRows();
-  size_t Nn = NodeMatrix->getLocalNumRows();
+  // size_t Nn = NodeMatrix->getLocalNumRows();
```
Comment (jhux2) [inline packages/muelu/src/Transfers/Smoothed-Aggregation/MueLu_ReitzingerPFactory_def.hpp:180]: just delete?

File: packages/pamgen/test/rtc_test/rtcompiler-test.cc [hunk 2]
```diff
@@ -267,7 +267,7 @@ int main( int argc, char* argv[] )
   realProgram(arrayOneReal, arrayTwoReal, plainValOneReal,
               plainValTwoReal, plainValThreeReal, testChar);
 
-  const double epsilon = 1.0e-15;
+  [[maybe_unused]] const double epsilon = 1.0e-15;
```
Comment (jhux2) [inline packages/pamgen/test/rtc_test/rtcompiler-test.cc:270]: I didn't know about this.

File: packages/panzer/dof-mgr/test/cartesian_topology/tCartesianDOFMgr_DynRankView.cpp [hunk 3]
```diff
@@ -70,7 +70,7 @@ TEUCHOS_UNIT_TEST(tCartesianDOFMgr_DynRankView, threed)
   #endif
 
   int np = comm.getSize(); // number of processors
-  int rank = comm.getRank(); // processor rank
+  // int rank = comm.getRank(); // processor rank
```
Comment (jhux2) [inline packages/panzer/dof-mgr/test/cartesian_topology/tCartesianDOFMgr_DynRankView.cpp:73]: why not delete

File: packages/panzer/dof-mgr/test/cartesian_topology/tCartesianDOFMgr_DynRankView.cpp [hunk 4]
```diff
@@ -109,8 +109,8 @@ TEUCHOS_UNIT_TEST(tCartesianDOFMgr_DynRankView, threed)
   dofManager->addField("eblock-0_1_0","UZ",pattern_U);
 
   // int temp_num = dofManager->getFieldNum("TEMPERATURE");
-  int ux_num   = dofManager->getFieldNum("UX");
-  int uy_num   = dofManager->getFieldNum("UY");
+  // int ux_num   = dofManager->getFieldNum("UX");
+  // int uy_num   = dofManager->getFieldNum("UY");
```
Comment (jhux2) [inline packages/panzer/dof-mgr/test/cartesian_topology/tCartesianDOFMgr_DynRankView.cpp:113]: why not delete

**Review by jhux2 — COMMENTED** [review]
See my comments

File: packages/pamgen/test/rtc_test/rtcompiler-test.cc (hunk 2 above)
Comment (cgcgcg) [reply packages/pamgen/test/rtc_test/rtcompiler-test.cc:270]: Let's see if that compiles everywhere...

**Review by jhux2 — APPROVED** [review]
keen

File: packages/panzer/dof-mgr/src/Panzer_GlobalIndexer.hpp [hunk 5]
```diff
@@ -255,8 +255,8 @@ class GlobalIndexer {
        functor.num_dofs = lids.extent(1);
 
 #ifdef PANZER_DEBUG
-     TEUCHOS_ASSERT(functor.local_lids.extent(1) >= num_dofs);
-     TEUCHOS_ASSERT(functor.global_lids.extent(1) >= num_dofs);
+     TEUCHOS_ASSERT(functor.local_lids.extent_int(1) >= num_dofs);
+     TEUCHOS_ASSERT(functor.global_lids.extent_int(1) >= num_dofs);
```
Comment (rppawlo) [inline packages/panzer/dof-mgr/src/Panzer_GlobalIndexer.hpp:259]: I would avoid using the extent_int() functions in kokkos views. I think they plan to deprecate them eventually. Maybe just use a cast instead?

**Review by jhux2 — APPROVED** [review]
lgtm

**Comment by jhux2** [comment]
@trilinos/framework It's unclear why the AT failed this PR.  There are no errors showing on the dashboard for the last few rounds of testing.

**Pull request merged**